node/internal/node_utils.bzl

```diff
@@ -17,27 +17,44 @@ node_library(
 
 def init_module(repository_ctx, module):
     targets = [struct(module = module, path = module.basename)]
+    # Extract nested targets within wrapped node_modules directories
     wrapped_paths = execute(repository_ctx, [
         "/bin/sh", "-c",
         "(find %s/node_modules -type d | grep '^%s\(/node_modules/[^/]\{1,\}\)\{1,\}$') 2>/dev/null || true" % (module, module),
     ]).stdout.split("\n")
     repo_prefix_len = len(str(repository_ctx.path(".")))
+
     for w in [repository_ctx.path(w[repo_prefix_len+1:]) for w in wrapped_paths if w and not w.endswith(".bin")]:
-        targets.append(struct(module = w, path = str(w)[repo_prefix_len+14:]))
+        if w.basename.startswith("@"):
+            for scoped_module in w.readdir():
+                mangled_package_name = mangle_package_name(w.basename, scoped_module.basename)
+                mangled_module_path = "%s/%s" % (w.dirname, mangled_package_name)
+                # mangled_module_path = "%s/%s__SLASH__%s" % (w.dirname, w.basename.replace("@", "__AT__"), scoped_module.basename)
+                execute(repository_ctx, ["mv", scoped_module, mangled_module_path])
+                renamed_module = repository_ctx.path(w.dirname).get_child(mangled_package_name)
+                # renamed_module = repository_ctx.path(w.dirname).get_child("%s__SLASH__%s" % (w.basename.replace("@", "__AT__"), scoped_module.basename))
+                targets.append(struct(module = renamed_module, path = str(repository_ctx.path(mangled_module_path))[repo_prefix_len+14:]))
+        else:
+            targets.append(struct(module = w, path = str(w)[repo_prefix_len+14:]))
+
     for t in targets:
         deps_cmd = [
             repository_ctx.path(repository_ctx.attr._node),
             repository_ctx.path(repository_ctx.attr._deps),
             t.module,
         ] + repository_ctx.attr.indeps.keys()
         deps = execute(repository_ctx, deps_cmd).stdout.strip()
+
         indeps = ["//%s:node_module" % (i) for i in repository_ctx.attr.indeps.get(t.module.basename, [])]
 
         wrapped_deps = []
         wrapped_path = t.module.get_child("node_modules")
         if wrapped_path.exists:
             for sub_module in wrapped_path.readdir():
                 if sub_module.basename.startswith("."): continue
+                # scoped modules aren't actual modules so don't include them
```

Comment:
Does this mean we don't support scoped modules as wrapped dependencies?

Reply:
No. We're just skipping over the directory that contained them before they were renamed and removed. [Here](https://github.com/happy-co/rules_node/pull/1/files/86a6cb6a129b0288bed0ceeec51ac344725b9dea#diff-967bff59cbc0d8ceaeacdc8d8d03f7f6R28)